src/traces/waterfall/cross_trace_calc.js

```diff
@@ -0,0 +1,50 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Registry = require('../../registry');
+var barCrossTraceCalc = require('../bar/cross_trace_calc').crossTraceCalc;
+
+module.exports = function crossTraceCalc(gd, plotinfo) {
+
+    barCrossTraceCalc(gd, plotinfo);
+
+    var xa = plotinfo.xaxis;
+    var ya = plotinfo.yaxis;
+
+    var fullTraces = gd._fullData;
+    var calcTraces = gd.calcdata;
+    var waterfalls = [];
+    var i;
+
+    for(i = 0; i < fullTraces.length; i++) {
+        var fullTrace = fullTraces[i];
+        if(
+            fullTrace.visible === true &&
+            Registry.traceIs(fullTrace, 'bar') &&
```

Comment:
We can :zap: this line. The `fullTrace.type === 'waterfall'` check below should suffice.